### PR TITLE
some work towards support of small IDEs

### PR DESCRIPTION
**Pull request by devoncarew**
- some work towards support of small IDEs

This adds a new project wizard contribution for small IDEs. It also does some cleanup where we display a combobox of SDKs - we now populate and maintain the SDK history (list of previously selected SDKs). This PR does not change our supported IntelliJ platforms - we want to set up some testing on the bots per platform when we do add support.

@pq

### CODE REVIEW

File: src/io/flutter/actions/DeviceSelectorAction.java [hunk 1]
```diff
@@ -109,7 +109,7 @@ public void deviceRemoved(ConnectedDevice device) {
   }
 
   private void updateVisibility(final Project project, final Presentation presentation) {
-    ApplicationManager.getApplication().invokeAndWait(() -> {
+    FlutterUtils.invokeAndWait(() -> {
```
Comment (pq) [inline src/io/flutter/actions/DeviceSelectorAction.java]: Good idea! 👍

File: resources/META-INF/plugin.xml [hunk 2]
```diff
@@ -172,8 +174,14 @@
     <!-- configurationType implementation="io.flutter.run.bazel.FlutterBazelRunConfigurationType"/>
     <programRunner implementation="io.flutter.run.bazel.FlutterBazelRunner"/ -->
 
+    <!-- IDEA only -->
     <moduleType id="FLUTTER_MODULE_TYPE" implementationClass="io.flutter.module.FlutterModuleType"/>
```
Comment (pq) [inline resources/META-INF/plugin.xml]: Maybe this should move to `idea-contribs.xml`?

File: src/io/flutter/module/FlutterSmallIDEProjectGenerator.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import io.flutter.FlutterErrors;
+import org.jetbrains.annotations.NotNull;
+
+import javax.swing.*;
+
+public class FlutterSmallIDEProjectGenerator extends WebProjectTemplate<String> {
+  @NotNull
+  @Override
+  public String getName() {
+    return "Flutter";
```
Comment (pq) [inline src/io/flutter/module/FlutterSmallIDEProjectGenerator.java]: Should this String be externalized?

File: src/io/flutter/sdk/FlutterSdkManager.java [hunk 4]
```diff
@@ -20,26 +21,20 @@
  * Monitors the application library table to notify clients when Flutter SDK configuration changes.
  */
 public class FlutterSdkManager {
-
-  private static FlutterSdkManager INSTANCE;
   private final EventDispatcher<Listener> myDispatcher = EventDispatcher.create(Listener.class);
   private final LibraryTableListener myLibraryTableListener = new LibraryTableListener();
   private boolean isFlutterConfigured;
 
+  public static FlutterSdkManager getInstance() {
```
Comment (pq) [inline src/io/flutter/sdk/FlutterSdkManager.java]: Are we expecting this to be non-null at call sites (or do we want to be able to?)  If so, maybe add `@NotNull`?

File: src/io/flutter/actions/DeviceSelectorAction.java (hunk 1 above)
Comment (devoncarew) [inline src/io/flutter/actions/DeviceSelectorAction.java]: This change lets us run on the current Android Studio (the dev channel build).

File: src/io/flutter/sdk/FlutterSdkManager.java (hunk 4 above)
Comment (devoncarew) [inline src/io/flutter/sdk/FlutterSdkManager.java]: Yup; added.

File: resources/META-INF/plugin.xml (hunk 2 above)
Comment (devoncarew) [inline resources/META-INF/plugin.xml]: sgtm, will move the `FlutterModuleConfigurationEditorProvider`. Off-hand, do you know why the four `editorNotificationProvider` are in the IDEA-only file?

**Pull request merged**